### PR TITLE
Custom user agent append in WebView

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
# Description
* Custom User-agent can be specified in the plist using **HostAppUserAgentInfo** key and the same will be appended in the end of the WebView user agent
* Updated Readme

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/Display/MiniAppWebView.swift [hunk 1]
```diff
@@ -12,5 +13,13 @@ internal class MiniAppWebView: WKWebView {
         self.init(frame: .zero, configuration: config)
         contentMode = .scaleToFill
         load(urlRequest)
+        evaluateJavaScript("navigator.userAgent") { [weak self] (result, error) in
+            if error != nil {
+                return
+            }
+            if let userAgent = result as? String {
+                self?.customUserAgent = userAgent + " " + environment.hostAppUserAgentInfo
```
Comment (corycaywood) [inline MiniApp/Classes/Display/MiniAppWebView.swift:21]: Will this set an extra space on the end of the user agent string even if `environment.hostAppUserAgentInfo` is not set? Can we check if `environment.hostAppUserAgentInfo` is empty before running the `evaluateJavascript` above?

I'm not sure if adding an extra space really matters, but it's better to not modify the user agent at all if there's no custom string to append.

File: MiniApp/Classes/Display/MiniAppWebView.swift [hunk 2]
```diff
@@ -12,5 +13,13 @@ internal class MiniAppWebView: WKWebView {
         self.init(frame: .zero, configuration: config)
         contentMode = .scaleToFill
         load(urlRequest)
+        evaluateJavaScript("navigator.userAgent") { [weak self] (result, error) in
```
Comment (corycaywood) [inline MiniApp/Classes/Display/MiniAppWebView.swift:16]: Ah, too bad there's no way to access this from swift 😣  Android lets you retrieve it from the webview settings.

Comment (ts-nitish-a-jain) [reply MiniApp/Classes/Display/MiniAppWebView.swift:16]: Isn't this accessible?
https://developer.apple.com/documentation/webkit/wkwebview/1414950-customuseragent

File: README.md [hunk 3]
```diff
@@ -166,11 +171,28 @@ extension ViewController: MiniAppMessageProtocol {
...
+
+
 ## License
 
 See the *LICENSE* file for more info.
 
-## Changelog
+<div id="change-log"></div>
+
+# Changelog
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply README.md:195]: Oh yeah. Will add it 👍

File: README.md [hunk 4]
```diff
@@ -30,22 +40,13 @@ In order to run your MiniApp you will have to provide the following,
...
-To run the example project, clone the repo, and run `pod install` from the Example directory first.
-
-### Custom parameters
-
-In order to load your MiniApp, you will have to use your own Host App ID and your Subscription key. These can either be set in project configuration plist (`RASApplicationIdentifier`, `RASProjectSubscriptionKey`) or by taping the top right configuration icon in the example application. Also we don't currently host a public API, so you will need to provide your own Base URL for API requests by setting it in project configuration plist (`RMAAPIEndpoint`)
-
-### Testing mode
-
-The SDK can be configured to load MiniApps in 2 modes : Published mode and Testing mode. Published mode only allows access to the last published version of your MiniApps, whereas Testing mode allows you to load all your MiniApps versions, as long as they are in `Testing` state on RAS console. To configure the loading mode, you can either do it in the .plist file with the configuration variable `RMAIsTestMode`, or by taping the top right configuration icon in the example application.
-
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply README.md:48]: Yes, we have moved it below.

File: Example/Info.plist [hunk 5]
```diff
@@ -30,6 +30,8 @@
 	<string>RAS_SUBSCRIPTION_KEY</string>
 	<key>RMAAPIEndpoint</key>
 	<string>https://www.example.com</string>
+	<key>HostAppUserAgentInfo</key>
+	<string>HOSTAPPNAME_AND_VERSION</string>
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply Example/Info.plist:34]: @Climbatize The requirement is to provide a **Custom User agent** string by the Hosting application. If the requirement was to retrieve the host app name and version, that's the solution

File: MiniApp/Classes/Environment.swift [hunk 6]
```diff
@@ -50,6 +53,10 @@ internal class Environment {
         return bool(for: customIsTestMode, fallback: .isTestMode)
     }
 
+    var hostAppUserAgentInfo: String {
+        return value(for: .none, fallback: .hostAppUserAgentInfo)
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply MiniApp/Classes/Environment.swift:57]: Check my comment above regarding the Custom user agent string

File: Example/Info.plist [hunk 7]
```diff
@@ -30,6 +30,8 @@
 	<string>RAS_SUBSCRIPTION_KEY</string>
 	<key>RMAAPIEndpoint</key>
 	<string>https://www.example.com</string>
+	<key>HostAppUserAgentInfo</key>
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply Example/Info.plist:33]: 👍

File: MiniApp/Classes/Display/MiniAppWebView.swift (hunk 1 above)
Comment (ts-leojoseph-r) [reply MiniApp/Classes/Display/MiniAppWebView.swift:21]: Sure, will add it 👍

File: MiniApp/Classes/Display/MiniAppWebView.swift (hunk 2 above)
Comment (ts-leojoseph-r) [reply MiniApp/Classes/Display/MiniAppWebView.swift:16]: @ts-nitish-a-jain customUserAgent string will be empty in the first place. Check here https://developer.apple.com/forums/thread/106744

**Pull request merged**